src/medicalpage/SingleMedication.js

```diff
@@ -0,0 +1,150 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { EditStatusIcon } from './EditStatus';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import {
+  faTrash,
+  faEdit,
+  faCheckSquare
+} from '@fortawesome/free-solid-svg-icons';
+import Input from '../components/Input';
+
+library.add(faTrash, faEdit, faCheckSquare);
+
+const StyledInput = styled.input`
+  width: 100%;
+`;
+
+const SingleMedicationContainer = styled.li`
+  display: grid;
+  grid-template-rows: auto;
+  grid-template-columns: 1fr 3fr 2fr 1fr 1fr;
+  list-style: none;
+`;
+
+const TimeContainer = styled.div`
+  width: 100%;
+  grid-column: 1;
+`;
+
+const MedicineContainer = styled.div`
+  width: 100%;
+  grid-column: 2;
+`;
+
+const DosageContainer = styled.div`
+  width: 100%;
+  grid-column: 3;
+`;
+
+const DeleteIcon = styled(FontAwesomeIcon)`
+  grid-column: 4;
+`;
+
+const DeleteButton = styled.button`
+  grid-column: 4;
+`;
+const StyledEditStatusIcon = styled(EditStatusIcon)`
+  grid-column: 5;
+`;
+
+export default function SingleMedication({
+  singleMedication,
+  onSingleMedicationDelete,
+  onSingleMedicationChange
+}) {
+  const [isEditable, setIsEditable] = useState(false);
+  const { id, time, medicine, dosage } = singleMedication;
+
+  function getTimeIcon(time) {
+    switch (time) {
+      case 'morning':
+        return '🌘';
+      case 'midday':
+        return '🌕';
+      case 'evening':
+        return '🌑';
+      default:
+        return '💊';
+    }
+  }
+  function onDeleteBtnClick(event, onSingleMedicationDelete) {
```

Comment:
Possibly an `edit` and `delete` function could be extracted to `utils.js` if you are using them in similar ways throughout your app